src/contexts/MatchContext.tsx

```
@@ -0,0 +1,144 @@
+      updatedPairHistory[p4.id][p3.id] =
+        (updatedPairHistory[p4.id][p3.id] || 0) + 1;
+    });
+
+    setPairHistory(updatedPairHistory);
+  };
+
+  const completeMatches = () => {
+    const playersInMatches = matches.reduce(
+      (acc, match) => [...acc, ...match.team1, ...match.team2],
+      [] as Player[]
+    );
```

Biome

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)